lib/novelconverter.rb

```diff
@@ -593,32 +605,49 @@ def convert_main_for_text(text)
 
     @use_dakuten_font = @converter.use_dakuten_font
 
-    converted_text
+    [converted_text]
   end
 
   #
   # 管理小説変換時の実質的なメイン処理
   #
   # 引数 subtitles にデータを渡した場合はそれを直接使う
   # is_hotentry を有効にすると出力されるテキストファイルにあらすじや作品タイトル等が含まれなくなる
+  # また、 is_hotentry を有効にすると分割も行われなくなる
   #
   def convert_main_for_novel(subtitles = nil, is_hotentry = false)
     toc = Downloader.get_toc_data(@setting.archive_path)
     unless subtitles
       subtitles = cut_subtitles(toc["subtitles"])
     end
-    @converter.subtitles = subtitles
-    toc["story"] = @converter.convert(toc["story"], "story")
-    html = HTML.new
-    html.strip_decoration_tag = @setting.enable_strip_decoration_tag
-    site_setting = find_site_setting(toc["toc_url"])
-    html.set_illust_setting({current_url: site_setting["illust_current_url"],
-                             grep_pattern: site_setting["illust_grep_pattern"]})
-
-    sections = subtitles_to_sections(subtitles, html)
-    converted_text = create_novel_text_by_template(sections, toc, is_hotentry)
-
-    converted_text
+    if is_hotentry == false && @setting.slice_size > 0 && subtitles.length > @setting.slice_size
+      puts "#{@setting.slice_size}話ごとに分割して変換します"
+      array_of_subtitles = subtitles.each_slice(@setting.slice_size).to_a
+    else
+      array_of_subtitles = [subtitles]
+    end
+    array_of_converted_text = []
+    array_of_subtitles.each_with_index do |sliced_subtitles, index|
+      @converter.subtitles = sliced_subtitles
+      toc["story"] = @converter.convert(toc["story"], "story")
+      html = HTML.new
+      html.strip_decoration_tag = @setting.enable_strip_decoration_tag
+      site_setting = find_site_setting(toc["toc_url"])
+      html.set_illust_setting({ current_url: site_setting["illust_current_url"],
+                                grep_pattern: site_setting["illust_grep_pattern"] })
+
+      sections = subtitles_to_sections(sliced_subtitles, html)
+      array_of_converted_text.push(
+        create_novel_text_by_template(
+          sections, toc, is_hotentry,
+          array_of_subtitles.length == 1 ? nil : index + 1))
```

Comment:
__[RuboCop]__

Layout/MultilineMethodCallBraceLayout: Closing method call brace must be on the line after the last argument when opening brace is on a separate line from the first argument.


<!-- Comment by [SideCI](https://sideci.com) -->